kolibri/content/management/commands/setdirectory.py

```diff
@@ -0,0 +1,141 @@
+            self.stderr.write(self.style.ERROR(
+                "Cannot migrate content while Kolibri is running, please run:\n"
+                "\n"
+                "    kolibri stop\n"
+            ))
+            raise SystemExit()
+        except server.NotRunning:
+            pass
+
+        logger.info('Current content directory is {}'.format(current_directory))
+
+        dest_directory = os.path.join(destination, 'content')
+        logger.info('Migrating the content into {}'.format(dest_directory))
+        dest_exists = os.path.exists(dest_directory)
+
+        # Check if destination has content by checking if databases folder is not empty
+        # TODO: This check may not be sufficient
+        dest_database_directory = os.path.join(dest_directory, 'databases')
+        dest_content_exists = os.path.exists(dest_database_directory) and \
+            os.listdir(dest_database_directory)
+
+        # If destination has content inside
+        if dest_exists and dest_content_exists:
+            self.copy_when_dest_content_exists(dest_directory, current_directory)
+        else:
+            self.copy_when_dest_content_dne(dest_exists, dest_directory, current_directory)
+
+        rmtree(current_directory)
+        self.update_config_content_directory(dest_directory)
+
+    def copy_when_dest_content_exists(self, dest_directory, current_directory):
+        """
+        If destination has content inside, ask users if they want to keep the
+        content in the destination. Copy the content to destination depending
+        on the user response.
+        """
+        user_answer = input(
+            self.style.WARNING(
+                'The destination {} has content inside.\n'
+                'Do you want to keep the content? (Yes/No)'.format(dest_directory)))
+
+        # If the user wants to keep the content in the destination
+        if user_answer == 'yes' or user_answer == 'Yes':
+            self.copy_content(current_directory, dest_directory)
+
+        # If the user does not want to keep the content in the destination
+        elif user_answer == 'no' or user_answer == 'No':
+            self.copy_when_dest_content_dne(True, dest_directory, current_directory)
+        else:
+            self.stderr.write(
+                self.style.ERROR(
+                    '{} cannot be an answer to the question. '
+                    'Please answer \'Yes\' or \'No\'.'
+                    .format(user_answer)))
+            raise SystemExit()
+
+    def copy_when_dest_content_dne(self, dest_exists, dest_directory, current_directory):
+        """
+        If the destination defined by the user does not exist,
+        or the destination exists but does not have content inside,
+        or the destination exists and has content inside but the user does not
+        want to keep the content during the migration,
+        then remove the destination if exists and start the copying process.
+        """
```

Comment:
This function doesn't handle any user interaction and seems to just remove `dest_directory`, so perhaps the comment can be a bit more specific, or the order of calls and conditions can be re-arranged a bit?

Reply:
ah thank you! I just realized that after I refactored some of my code, this function becomes not really meaningful